src/compas/base.py

```diff
@@ -1,21 +1,28 @@
+"""
+If you ever feel tempted to use ABCMeta in your code: don't, just DON'T.
+Assigning __metaclass__ = ABCMeta to a class causes a severe memory leak/performance
+degradation on IronPython 2.7.
+
+See these issues for more details:
+ - https://github.com/compas-dev/compas/issues/562
+ - https://github.com/compas-dev/compas/issues/649
+"""
 from __future__ import print_function
 from __future__ import absolute_import
 from __future__ import division
 
-import abc
 import json
 from uuid import uuid4
 
 from compas.utilities import DataEncoder
 from compas.utilities import DataDecoder
-from compas.utilities import abstractclassmethod
-
-ABC = abc.ABCMeta('ABC', (object,), {'__slots__': ()})
 
 
 __all__ = [
     'Base',
 ]
+import abc
+ABC = abc.ABCMeta('ABC', (object,), {'__slots__': ()})
 
 
 class Base(ABC):
```

Comment:
Don't you think the other classes will be jealous that `Base` still inherits from `ABC`?

Reply:
they were indeed